solidity/contracts/converter/BancorConverterUpgrader.sol

```diff
@@ -230,9 +237,15 @@ contract BancorConverterUpgrader is IBancorConverterUpgrader, ContractRegistryCl
 
         for (uint16 i = 0; i < connectorTokenCount; i++) {
             address connectorAddress = _oldConverter.connectorTokens(i);
-            IERC20Token connector = IERC20Token(connectorAddress);
-            connectorBalance = connector.balanceOf(_oldConverter);
-            _oldConverter.withdrawTokens(connector, address(_newConverter), connectorBalance);
+            if (connectorAddress == address(0)) {
+                connectorBalance = etherToken.balanceOf(_oldConverter);
+                etherToken.withdrawTo(address(_newConverter), connectorBalance);
+            }
+            else {
+                IERC20Token connector = IERC20Token(connectorAddress);

```

Comment:
if the connector is EtherToken, we need to withdraw the ETH from it and send it to the new converter

Reply:
@yudilevi : The connector is not EtherToken at this point, it is the zero address.